AnswerKingAPI.yml

```diff
@@ -2,15 +2,230 @@ openapi: 3.0.1
+        '400':
+          description: Invalid parameters are provided.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+    get:
+      tags:
+        - Tags
+      summary:  List All tags.
+      operationId: listAllTags
+      responses:
+        '200':
+          description:  OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  items:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Tag.Response'
+  /api/tags/{id}:
+    parameters:
+      - schema:
+          type: string
+        name: id
+        in: path
+        required: true
+    get:
+      tags:
+        - Tags
+      summary: View Tag.
+      operationId: viewTag
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Tag.Response'
+        '400':
+          description: Invalid parameters are provided.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '404':
+          description: Tag with the given id does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+    put:
+      tags:
+        - Tags
+      summary: Modify Tag.
+      operationId:  modifyTag
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Tag.Request'
+      responses:
+        '200':
+          description:  OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Tag.Response'
+        '400':
+          description: Invalid parameters are provided.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '404':
+          description: Tag with the given id does not exist.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+    delete:
+      tags:
+        - Tags
+      summary:  Delete Tag.
+      operationId:  deleteTag
+      responses:
+        '204':
+          description:  No Content
+        '400':
+          description: Invalid parameters are provided.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '404':
+          description: Tag with the given id does not exist.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '410':
```

Comment:
There seems to be an inconsistency here with the deletion of tags. Adding the possibility of a 410 response implies that we can retire tags, however, I'm not seeing 410 responses added to our other endpoints (notably modifyTag, viewProductsForTag and assignTagToProducts, as well as possibly removeTagFromProducts if the product or tag is retired)

I've started a discussion on the bench-king-senior-dev-board Slack channelabout this.

Reply:
as responded in my comment in #bench-king-senior-dev-board group, I'd delete it for now until we discuss the whole "Retire" concept for all the endpoints.